src/actions/Bond.actions.ts

```diff
@@ -163,36 +222,37 @@ export const calculateUserBondDetails =
     let interestDue, pendingPayout, bondMaturationBlock;
 
     const bondDetails = await bondContract.bondInfo(address);
-    interestDue = bondDetails.payout / Math.pow(10, 9);
+    interestDue = toNum(bondDetails.payout) / Math.pow(10, 9);
     bondMaturationBlock = +bondDetails.vesting + +bondDetails.lastBlock;
     pendingPayout = await bondContract.pendingPayoutFor(address);
 
-    let allowance,
-      balance = 0;
+    const bondAddresses = addresses[networkID].BONDS as Nested;
+
+    let allowance, balance;
     if (bond === BONDS.ohm_dai) {
-      allowance = await reserveContract.allowance(address, addresses[networkID].BONDS.OHM_DAI);
+      allowance = await reserveContract.allowance(address, bondAddresses.OHM_DAI);
 
-      balance = await reserveContract.balanceOf(address);
+      balance = toNum(await reserveContract.balanceOf(address));
       balance = ethers.utils.formatUnits(balance, "ether");
     } else if (bond === BONDS.dai) {
-      allowance = await reserveContract.allowance(address, addresses[networkID].BONDS.DAI);
+      allowance = await reserveContract.allowance(address, bondAddresses.DAI);
 
-      balance = await reserveContract.balanceOf(address);
+      balance = toNum(await reserveContract.balanceOf(address));
       balance = ethers.utils.formatEther(balance);
     } else if (bond === BONDS.ohm_frax) {
-      allowance = await reserveContract.allowance(address, addresses[networkID].BONDS.OHM_FRAX);
+      allowance = await reserveContract.allowance(address, bondAddresses.OHM_FRAX);
 
-      balance = await reserveContract.balanceOf(address);
+      balance = toNum(await reserveContract.balanceOf(address));
       balance = ethers.utils.formatUnits(balance, "ether");
     } else if (bond === BONDS.frax) {
-      allowance = await reserveContract.allowance(address, addresses[networkID].BONDS.FRAX);
+      allowance = await reserveContract.allowance(address, bondAddresses.FRAX);
 
-      balance = await reserveContract.balanceOf(address);
+      balance = toNum(await reserveContract.balanceOf(address));
```

Comment:
Do we need `toNum` everywhere? it looks like the default type was BigNumber?

Reply:
yeah, because we are currently doing mathematical operations in this way: `bigNum + anotherBigNum` and TypeScript complains. the correct syntax for mathematical operations w/ BigNumber is `bigNum.add(anotherBigNum)`, but I wasn't sure if this would lead to further bugs.